utilities/ImpactUtilities.groovy

```diff
@@ -298,6 +308,110 @@ def scanOnlyStaticDependencies(List buildList, RepositoryClient repositoryClient
 	}
 }
 
+/*
+ * Method to query the DBB collections with a list of changed files  
+ * Configured through reportExternalImpacts* build properties
+ */
+
+def reportExternalImpacts(RepositoryClient repositoryClient, Set<String> changedFiles){
+	// query external collections to produce externalImpactList
+
+	Map<String,HashSet> collectionImpactsSetMap = new HashMap<String,HashSet>() // <collection><List impactRecords>
+	List<Pattern> collectionMatcherPatterns = createMatcherPatterns(props.reportExternalImpactsCollectionPatterns)
+
+	// caluclated and collect external impacts
+	changedFiles.each{ changedFile ->
+
+		List<PathMatcher> fileMatchers = createPathMatcherPattern(props.reportExternalImpactsAnalysisFileFilter)
+		
+		if(matches(changedFile, fileMatchers)){
+
+			if (props.reportExternalImpactsAnalysisDepths == "simple"){
+				// Simple resolution without recursive resolution
+				String memberName = CopyToPDS.createMemberName(changedFile)
+
+				def ldepFile = new LogicalDependency(memberName, null, null);
+				repositoryClient.getAllCollections().each{ collection ->
+					String cName = collection.getName()
+					if(matchesPattern(cName,collectionMatcherPatterns)){ // find matching collection names
+						if (cName != props.applicationCollectionName && cName != props.applicationOutputsCollectionName){
+							def Set<String> externalImpactList = collectionImpactsSetMap.get(cName) ?: new HashSet<String>()
+							def logicalFiles = repositoryClient.getAllLogicalFiles(cName, ldepFile);
+							logicalFiles.each{ logicalFile ->
+								def impactRecord = "${logicalFile.getLname()} \t ${logicalFile.getFile()} \t ${cName}"
```

Comment:
Depending on the length of each item using tab may result in a file difficult to read in text mode. But I don't know what to use. Why not produce a json file ?

Reply:
Idea of using a text file is to use this as an build file within the external repositories